### ISSUE TITLE
2.3.0 test failures (test_get_data_dir__missing & test_get_data_dir__from_path)

### ISSUE THREAD

**Issue by sebastic** [pyproj4/pyproj#400]
The Debian package build for python-pyproj 2.3.0 failed on Debian unstable (with proj 6.1.1 from experimental) due to test failures:
```
I: pybuild base:217: cd /build/python-pyproj-2.3.0+ds/.pybuild/cpython3_3.7_pyproj/build; python3.7 -m pytest test
============================= test session starts ==============================
platform linux -- Python 3.7.4, pytest-4.6.5, py-1.8.0, pluggy-0.12.0
rootdir: /build/python-pyproj-2.3.0+ds
plugins: cov-2.7.1
collected 251 items

test/test_awips221.py .                                                  [  0%]
test/test_crs.py ....................................................... [ 22%]
.................................................                        [ 41%]
test/test_crs_cf.py ................                                     [ 48%]
test/test_datadir.py F.....F..                                           [ 51%]
test/test_datum.py .                                                     [ 52%]
test/test_datum_shift.py ....                                            [ 53%]
test/test_doctest_wrapper.py .                                           [ 54%]
test/test_exception_logging.py ..                                        [ 54%]
test/test_geod.py ............................                           [ 66%]
test/test_list.py .....                                                  [ 68%]
test/test_pickle.py .                                                    [ 68%]
test/test_proj.py ........................                               [ 78%]
test/test_show_versions.py ...                                           [ 79%]
test/test_transform.py .                                                 [ 79%]
test/test_transformer.py ............................................... [ 98%]
....                                                                     [100%]

=================================== FAILURES ===================================
__________________________ test_get_data_dir__missing __________________________

    @unittest.skipIf(os.name == "nt", reason="Cannot modify Windows environment variables.")
    def test_get_data_dir__missing():
        with proj_env(), pytest.raises(DataDirError), patch(
            "pyproj.datadir.os.path.abspath", return_value="INVALID"
        ), patch("pyproj.datadir.find_executable", return_value=None):
            unset_data_dir()
            os.environ.pop("PROJ_LIB", None)
>           assert get_data_dir() is None
E           AssertionError: assert '/usr/share/proj' is None
E            +  where '/usr/share/proj' = get_data_dir()

test/test_datadir.py:63: AssertionError
_________________________ test_get_data_dir__from_path _________________________

    @unittest.skipIf(os.name == "nt", reason="Cannot modify Windows environment variables.")
    def test_get_data_dir__from_path():
        with proj_env(), temporary_directory() as tmpdir, patch(
            "pyproj.datadir.os.path.abspath", return_value="INVALID"
        ), patch("pyproj.datadir.find_executable") as find_exe:
            unset_data_dir()
            os.environ.pop("PROJ_LIB", None)
            find_exe.return_value = os.path.join(tmpdir, "bin", "proj")
            proj_dir = os.path.join(tmpdir, "share", "proj")
            os.makedirs(proj_dir)
            create_projdb(proj_dir)
>           assert get_data_dir() == proj_dir
E           AssertionError: assert '/usr/share/proj' == '/tmp/tmp24f36zqv/share/proj'
E             - /usr/share/proj
E             + /tmp/tmp24f36zqv/share/proj

test/test_datadir.py:140: AssertionError
==================== 2 failed, 249 passed in 16.04 seconds =====================
E: pybuild pybuild:341: test: plugin distutils failed with: exit code=1: cd /build/python-pyproj-2.3.0+ds/.pybuild/cpython3_3.7_pyproj/build; python3.7 -m pytest test
```

The package builds sets `PROJ_DIR` & `PROJ_LIB` to use the system provided proj:
```
export PROJ_DIR=/usr
export PROJ_LIB=/usr/share/proj
```

These tests should probably be skipped when the data dir is already set.

**Comment by snowman2** [comment]
Thanks for the report. I really should ping you when I am thinking about doing a release to have you run some builds and ensure I didn't break anything :laughing:.

In think the solution here is to mock `sys.prefix` for those tests to return an invalid string.

**Comment by sebastic** [comment]
> I really should ping you when I am thinking about doing a release to have you run some builds and ensure I didn't break anything

Publishing pre-releases (beta and/or RCs) is a good idea to catch these kind of issues before the final release.

**Comment by sebastic** [comment]
Confirmed fixed with the changes from 288c149f5306e8b62ee2652bfe0ac5efe555f681

**Comment by snowman2** [comment]
Good to hear 👍
